core/src/main/java/kafka/server/SharePartition.java

```diff
@@ -1174,6 +1197,52 @@ private void releaseAcquisitionLockOnTimeout(String memberId, long firstOffset,
         }
     }
 
+
+    // Visible for testing
+     boolean isWriteShareGroupStateSuccessful(List<PersisterStateBatch> stateBatches) {
+        try {
+            // Persister class can be null during active development and shall be driven by temporary config.
+            if (persister == null)
+                return true;
+            WriteShareGroupStateResult response = persister.writeState(new WriteShareGroupStateParameters.Builder()
+                .setGroupTopicPartitionData(new GroupTopicPartitionData.Builder<PartitionStateBatchData>()
+                    .setGroupId(this.groupId)
+                    .setTopicsData(Collections.singletonList(new TopicData<>(topicIdPartition.topicId(),
+                        Collections.singletonList(PartitionFactory.newPartitionStateBatchData(
+                            topicIdPartition.partition(), ++stateEpoch, startOffset, stateBatches))))
```

Comment:
`++stateEpoch` is wrong. It particular, it should not be incremented on each call. The usual behaviour is that it will be initialized from the persister when the SharePartition comes into existence, and then the same value should be used on all writes. We could consider hiding it entirely in the persister actually, but for now, I suggest initializing a member variable `stateEpoch` to zero, and then just passing that value on all calls to the persister which need a state epoch.